src/main/java/hu/eszter/bokkon/model/participants/Farmer.java

```diff
@@ -53,18 +63,40 @@ public void addAnimals(Animal animal, int howMany) {
      */
     @Override
     public void removeAnimals(Animal animal, int howMany) {
-        if (farmerLiveStock.get(animal) >= howMany) {
-            farmerLiveStock.computeIfPresent(animal, (k, v) -> v - howMany);
-        } else if (farmerLiveStock.get(animal) == 0) {
-            farmerLiveStock.remove(animal);
-        } else {
+        if (animalStock.get(animal) >= howMany) {
+            animalStock.computeIfPresent(animal, (k, v) -> v - howMany);
+        }  else {
             System.out.println("No animal was removed from " + name + "'s stock.");
         }
     }
 
+    public void setOneAnimalCountToZero(Animal animalToSetZero) {
+        animalStock.put(animalToSetZero, 0);
+    }
+
+    public void setAnimalsCountToZero() {
+        setOneAnimalCountToZero(new Rabbit());
+        setOneAnimalCountToZero(new Sheep());
+        setOneAnimalCountToZero(new Pig());
+        setOneAnimalCountToZero(new Cow());
+        setOneAnimalCountToZero(new BigDog());
+
+    }
+
     public Animal rollDice(Die dice) {
```

Comment:
rolling and the random-generator itself should be moved to Die, Game or Util, i.e. to a more centric place. Farmer, of course can still call it